src/Controller/VillaController.php

```diff
@@ -25,12 +34,34 @@ public function index(VillaRepository $villaRepository): Response
 
 
     /**
-     * @Route("/{id}", name="villa_show", methods={"GET"})
+     * @Route("/{id}", name="villa_show", methods={"GET","POST"})
      */
-    public function show(Villa $villa): Response
+    public function show(Villa $villa, Request $request, MailerInterface $mailer): Response
     {
+        $booking= new Booking();
+        $form = $this->createForm(BookingType::class, $booking);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $information = $form->getData();
```

Comment:
The idea of using a class behind the form is to use the object to get information instead of doing a getData() to get raw data.
So you should delete the $information variable and use the $booking instead